### PR TITLE
THEEDGE-1928: Add devicegroups/devices many2many back reference

### PR DESCRIPTION
**Pull request by thom-at-redhat**
# Description

Add many2many back reference for devicegroups/devices.  Incorporates Ana"s changed from PR 86.  Still need to resolve local "go test" issue that works elsewhere

Fixes # THEEDGE-1928

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/services/devices.go [hunk 1]
```diff
@@ -103,6 +103,8 @@ func (s *DeviceService) GetDeviceByID(deviceID uint) (*models.Device, error) {
 		s.log.WithField("error", result.Error.Error()).Error("Error finding device")
 		return nil, new(DeviceNotFoundError)
 	}
+	//Load from device DB the groups info and add to the new struct
+	db.DB.Model(&device).Association("DevicesGroups").Find(&device.DevicesGroups)
```
Comment (acosferreira) [inline pkg/services/devices.go:107]: Add a error handling please.

File: pkg/services/devices.go [hunk 2]
```diff
@@ -116,6 +118,8 @@ func (s *DeviceService) GetDeviceByUUID(deviceUUID string) (*models.Device, erro
...
```
Comment (acosferreira) [inline pkg/services/devices.go:122]: Add a error handling please.

File: pkg/services/devices.go [hunk 3]
```diff
@@ -384,13 +391,20 @@ func (s *DeviceService) GetDevices(params *inventory.Params) (*models.DeviceDeta
 		if !ok {
 			dbDeviceID = 0
 		}
+
+		//Load from device DB the groups info and add to the new struct
+		var storeDevice models.Device
+		db.DB.Where("id=?", dbDeviceID).First(&storeDevice)
```
Comment (acosferreira) [inline pkg/services/devices.go:397]: Add a error handling please.

**Review by acosferreira — CHANGES_REQUESTED** [review]

**Review by loadtheaccumulator — APPROVED** [review]
lgtm

**Pull request merged**